practice/flutter/flutter/views.py

```diff
@@ -0,0 +1,129 @@
+    sorted_flutts_all = logic.sort_flutts_by_most_recent(all_flutts)
+    username_display = logic.get_username_display(request.user.username)
+    arguments = {
+        'last_ten_flutts': sorted_flutts_all[:10],
+        'username_display': username_display
+    }
+    return render(request, 'flutter/index.html', arguments)
+
+
+@login_required(login_url="login")
+def render_post_form(request):
+    """Renders the form for posting a Flutt.
+
+    Requires login by the user, will direct someone who is not logged in to login page.
+    """
+    username_display = logic.get_username_display(request.user.username)
+    arguments = {
+        'username_display': username_display
+    }
+    return render(request, 'flutter/post_form.html', arguments)
+
+
+def render_post_ack(request):
+    """Receives Flutt data, creates flutt and saves, and acknowledges receipt.
+
+    Returns a 400 response if the user did not add something to the body of the Flutt.
+    """
+    username = request.user.username
+    body = request.POST['body']
+    time = logic.get_current_time()
+    user_id = request.user.id
+    try:
+        logic.create_and_save_flutt(username, body, time, user_id)
+    except ValueError:
+        return HttpResponse('You forgot to write something! <a href="/post">Please try again</a>', status=400)
+    username_display = logic.get_username_display(user.username)
+    arguments = {
+        'username_display': username_display
+    }
+    return render(request, 'flutter/post_ack.html', arguments)
+
+
+def render_search(request):
+    """Detects what kind of search the user is attempting (by user or by text).
+
+    If it is by user, gets the user ID of the searched user, then directs to the 'search_by_user_id' url. If there is no
+    user by that name, or if they haven't posted a Flutt, returns a 400 response.
+
+    If it is by text, completes the search and returns 10 matches sorted by most recent. If there are no results,
+    returns a 400 response.
+    """
+    if 'searchtext' in request.GET:
+        search_text = request.GET.get('searchtext', False)
+        try:
+            matches = logic.get_matches_by_search_text(search_text)
+        except LookupError:
+            return HttpResponse('No results for that search text. <a href="/">Please try again.<a>', status=400)
+        sorted_flutts_by_search_text = logic.sort_flutts_by_most_recent(matches)
+        username_display = logic.get_username_display(request.user.username)
+        arguments = {
+            'last_ten_flutts': sorted_flutts_by_search_text[:10],
+            'username_display': username_display
+        }
+        return render(request, 'flutter/index.html', arguments)
+    elif 'searchuser' in request.GET:
+        username = request.GET['searchuser']
+        try:
+            user_id = logic.get_user_id(username)
+        except LookupError:
+            return HttpResponse('Sorry, that user doesn\'t exist. <a href="/">Please try again.</a>', status=400)
```

Comment:
If you want to display a special message when no search results, you can do that in the template with an if tag.